poor/util.py

```diff
@@ -161,67 +161,79 @@ def find_closest(xs, ys, x, y, subset=None):
             min_dist = dist
     return min_index
 
-def format_distance(meters, n=2):
+def format_distance(meters, n=2, voice=False):
     """Format `meters` to `n` significant digits and unit label."""
     if poor.conf.units == "american":
         feet = 3.28084 * meters
-        return format_distance_american(feet, n)
+        return format_distance_american(feet, n, voice)
     if poor.conf.units == "british":
         yards = 1.09361 * meters
-        return format_distance_british(yards, n)
-    return format_distance_metric(meters, n)
+        return format_distance_british(yards, n, voice)
+    return format_distance_metric(meters, n, voice)
 
-def format_distance_american(feet, n=2):
+def format_distance_american(feet, n=2, voice=False):
     """Format `feet` to `n` significant digits and unit label."""
     if (n > 1 and feet >= 1000) or feet >= 5280:
         distance = feet / 5280
-        units = "mi"
+        if voice: units = _("miles")
+        else: units = "mi"
+        in_feet = False
     else:
         # Let's not use units less than a foot.
         distance = feet
-        units = "ft"
+        if voice: units = _("feet")
+        else: units = "ft"
+        in_feet = True
     ndigits = n - math.ceil(math.log10(abs(max(1, distance)) + 1/1000000))
-    if units == "ft":
+    if in_feet:
         ndigits = min(0, ndigits)
     distance = round(distance, ndigits)
     fstring = "{{:.{:d}f}} {{}}".format(max(0, ndigits))
     return fstring.format(distance, units)
```

Comment:
I am sure it will not make a big difference, but I would expect the current code to be a bit faster. Or should I still replace it according to your comment?

Reply:
It's not about speed, but clarity. If you don't understand it, skip it, I'll fix it later.

Reply:
Started changing and ... it turned out not to work for km/m. Should I still replace it and make special version for metric system?

Reply:
Ah, yes, of course, maybe use `re.sub`:

```python
units = re.sub("^m$", _("meters"), units)
units = re.sub("^km$", _("kilometers"), units)
```

Reply:
Seems to be working now. Commit submitted